airbyte-webapp/STYLEGUIDE.md

```diff
@@ -70,3 +70,111 @@ Our SCSS color variables compile to `rgb(X, Y, Z)`, which is an invalid value in
+
+* `index.ts` - Used to the main component and supporting functions or types to the app.
+* `{MainComponentName}.tsx` - The main component, also exported through `index.ts`
+* `{MainComponentName}.test.tsx` - The test file
+* `{MainComponentName}.module.scss` - The main component's style file
+* Files for additional components that support the main Component directly, using the same naming conventions as above.
+* `types.ts` (optional) - Types shared between different sub-components
+* `utils.ts` (optional) - Any utilities that support the components
+
+When using supporting components, the folder could become rather full of them. In those cases Components should be split into their own folders parallel to the main component, especially when the supporting component also needs to be broken down into multiple sub components.
+
+Here's a hypothetical example: The app has a streams panel with a lot of sub-components including a streams table. Instead of placing all the components under `src/components/connection/StreamsPanel`, the table should be broken out into it's own sub-folder as a child of the `connection` folder.
+
+❌ Incorrect - While StreamsTable is only used in the StreamsPanel, it also has its own sub-components.
+
+```
+src/
+  components/
+    connection/
+      StreamsPanel/
+        index.ts
+        StreamsPanel.tsx
+        StreamsPanelHeader.tsx
+        StreamsPanelContent.tsx
+        StreamsTable.tsx
+        StreamsTableBody.tsx
+        StreamsTableCell.tsx
+        StreamsTableHeader.tsx
+        StreamsTableRow.tsx
+```
+
+✅ Correct - StreamsTable is separated into its own folder alongside StreamsPanel
+
+```
+src/
+  components/
+    connection/
+      StreamsPanel/
+        index.ts
+        StreamsPanel.tsx
+        StreamsPanelHeader.tsx
+        StreamsPanelContent.tsx
+      StreamsTable/
+        index.ts
+        StreamsTable.tsx
+        StreamsTableBody.tsx
+        StreamsTableCell.tsx
+        StreamsTableHeader.tsx
+        StreamsTableRow.tsx
+```
+
+### Pages Structure
+
+> The Airbyte team is currently restructuring how pages are organized in the codebase. This describes the ideal structure for the `pages/` folder.
+
+React components that represent pages and routes are split from the rest of the components in order to provide a clear way to identify the individual pages that are available in the app.
+
+Each sub-folder in `pages/` corresponds to the domain path in the url. For example. `/workspaces/{workspaceId}/connections/*` pages are available in `pages/connections/`.
+
+Pages are not nested within sub-folders. Instead, all pages are alongside each other and the nesting is configured in the routes file.
+
+The page React component is intended to put together a series of more heavy-hitting components. The components that support the page are part of the `components/` folder. Smaller components that support the page component can be created in the page's folder.
```

Comment:
Phrasing suggestion:

```suggestion
A page component should not contain any complex logic, it should rather focus on composing other components. The components that are used in the page belong in the `components/` folder. Smaller components that support the page component can be created in the page's folder.
```

> Smaller components that support the page component can be created in the page's folder.

I wonder if this is something we should encourage? The way this is currently phrased makes it unclear to me what types of components could live here. Personally I'm in favor of keeping _only_ pages in the `pages` folder. I think it's a "slippery slope" if we allow other random components to be co-located there, which leads to a lot of components not in the `components` folder (probably how we ended up where we are currently).

Reply:
Agree.